contracts/core/Registry.sol

```diff
@@ -62,9 +62,7 @@ contract Registry is IRegistry, Native, AccessControl, Transfer, Initializable,
     /// @notice Reverts UNAUTHORIZED() if the caller is not the profile owner
     /// @param _profileId The profile id
     modifier onlyProfileOwner(bytes32 _profileId) {
-        if (!_isOwnerOfProfile(_profileId, msg.sender)) {
-            revert UNAUTHORIZED();
-        }
+        _checkOnlyProfileOwner(_profileId);
```

Comment:
I don't think we need to do this, it is only used in this one place, so you are still following SOLID principles.

Reply:
Looks like I was late in explaining and showing the result of the change we can get, but anyways

Yeah, I know that internal function was used at just one place, but it's related to modifier which is kind of used up several times. Try to run some tests and we be getting a clear scenario.